### PR TITLE
Adds some minimal sensor data to the thermostat driver to allow basic…

### PR DESCRIPTION
**Pull request by lucboudreau**
… control by third party.

## Description:

**Related issue (if applicable):** fixes #<Tasmota issue number goes here>

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [ ] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [x] The code change is tested and works with Tasmota core ESP32 V.2.0.1.1
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/xdrv_39_thermostat.ino [hunk 1]
```diff
@@ -2022,8 +2022,16 @@ const char HTTP_THERMOSTAT_HL[]          PROGMEM = "{s}<hr>{m}<hr>{e}";
 
 #endif  // USE_WEBSERVER
 
-void ThermostatShow(uint8_t ctr_output)
+void ThermostatShow(uint8_t ctr_output, bool json)
 {
+  if (json) {
+    ResponseAppend_P(PSTR(",\"%s%i\":{"), "Thermostat", ctr_output); // this should probably stay static accross locales
+    ResponseAppend_P(PSTR("%s\"%s\":%i"), "", D_CMND_THERMOSTATMODESET, Thermostat[ctr_output].status.thermostat_mode);
```
Comment (arendst) [inline tasmota/xdrv_39_thermostat.ino:2029]: It. MUST be in english to satisfy JSON data handlers and was decided three years ago. Tasmota JSON is not localized.

Comment (lucboudreau) [reply tasmota/xdrv_39_thermostat.ino:2029]: And I agree. That's what I meant by static, and the code currently only uses fixed identifiers in English only.

😂

Comment (lucboudreau) [reply tasmota/xdrv_39_thermostat.ino:2029]: Let me know if I missed something, but it is already in English and will not get localized.

Comment (barbudor) [reply tasmota/xdrv_39_thermostat.ino:2029]: You can just put the `"Thermostat"` part into the format string.

Comment (lucboudreau) [reply tasmota/xdrv_39_thermostat.ino:2029]: @barbudor - 🤷‍♂️ Sure. Done.

File: tasmota/xdrv_39_thermostat.ino [hunk 2]
```diff
@@ -2022,8 +2022,16 @@ const char HTTP_THERMOSTAT_HL[]          PROGMEM = "{s}<hr>{m}<hr>{e}";
 
 #endif  // USE_WEBSERVER
 
-void ThermostatShow(uint8_t ctr_output)
+void ThermostatShow(uint8_t ctr_output, bool json)
 {
+  if (json) {
+    ResponseAppend_P(PSTR(",\"Thermostat%i\":{"), ctr_output);
+    ResponseAppend_P(PSTR("%s\"%s\":%i"), "", D_CMND_THERMOSTATMODESET, Thermostat[ctr_output].status.thermostat_mode);
+    ResponseAppend_P(PSTR("%s\"%s\":%f"), ",", D_CMND_TEMPTARGETSET, (Thermostat[ctr_output].temp_target_level / 10.0f));
```
Comment (arendst) [inline tasmota/xdrv_39_thermostat.ino:2030]: Tasmota does not support "%f". It uses too much code space. Instead use "%2_f". (In that case prefix the variable with an "&" too).
Look for example in the code.

Comment (lucboudreau) [reply tasmota/xdrv_39_thermostat.ino]: Fixed it. Thanks.

**Pull request merged**